src/sqlfluff/dialects/dialect_redshift.py

```diff
@@ -249,6 +250,34 @@ class CreateTableStatementSegment(BaseSegment):
     )
 
 
+@redshift_dialect.segment(replace=True)
+class InsertStatementSegment(BaseSegment):
+    """A `INSERT` statement.
+
+    Redshift has two versions of insert statements:
+        - https://docs.aws.amazon.com/redshift/latest/dg/r_INSERT_30.html
+        - https://docs.aws.amazon.com/redshift/latest/dg/r_INSERT_external_table.html
+    """
+
+    type = "insert_statement"
+    match_grammar = Sequence(
+        "INSERT",
+        "INTO",
+        Ref("TableReferenceSegment"),
+        OneOf(
+            OptionallyBracketed(Ref("SelectableGrammar")),
+            Sequence("DEFAULT", "VALUES"),
+            Sequence(
+                Ref("BracketedColumnReferenceListGrammar", optional=True),
```

Comment:
Did you work out which bit of this is causing the problem?

Reply:
Yeah, It was because I didn't have it as part of the `StatementSegment`. Once I did that, the clouds opened up :).

Reply:
That doesn't make sense though - as it is a segment you're replacing, it's already part of StatementSegment

Reply:
I removed that part. It's probably the fact that I added OneOf.